src/main/webapp/resources/js/contexts/AnalysisContext.js

```diff
@@ -51,12 +56,49 @@ const AnalysisContext = React.createContext(initialContext);
 
 function AnalysisProvider(props) {
   const [analysisContext, setAnalysisContext] = useState(initialContext);
+  const analysisIdentifier = window.location.pathname.match(/analysis\/(\d+)/)[1];
+
+  useEffect(() => {
+    getAnalysisInfo(analysisIdentifier).then(res => {
+      setAnalysisContext(analysisContext => {
+        return {
+          ...analysisContext,
+          analysis: res.analysis,
```

Comment:
This looks like it would be simpler just to destructor the response and just explicitly state the ones that are not. e.g 
```
return {...analysisContexxt, ...res, isCompleted: res.analysisState === "COMPLETED"} 
```

Reply:
I might even go as far as making the response from the server have all the variables set exactly as you want them (e.g. isCompleted, etc...) be set server side properly

Reply:
Updated in 8d63f91593b46a445bb918bea869f802e346b15f